src/main/java/org/testng/internal/annotations/IgnoreListener.java

```diff
@@ -0,0 +1,40 @@
+package org.testng.internal.annotations;
+
+import org.testng.annotations.ITestAnnotation;
+import org.testng.annotations.Ignore;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+
+public class IgnoreListener implements IAnnotationTransformer {
+
+    @Override
+    public void transform(ITestAnnotation annotation, Class testClass, Constructor testConstructor, Method testMethod) {
+        Class<?> typedTestClass = testClass;
+        if (testMethod != null) {
+            if (annotation.getEnabled()) {
+                ignoreTest(annotation, testMethod.getAnnotation(Ignore.class));
+            }
+            typedTestClass = testMethod.getDeclaringClass();
+        }
+        if (typedTestClass != null) {
+            if (annotation.getEnabled()) {
+                ignoreTest(annotation, typedTestClass.getAnnotation(Ignore.class));
+            }
+            if (annotation.getEnabled() && typedTestClass.getPackage() != null) {
+                ignoreTest(annotation, typedTestClass.getPackage().getAnnotation(Ignore.class));
+            }
+        }
+    }
+
+    private static void ignoreTest(ITestAnnotation annotation, Ignore ignore) {
+        if (ignore != null) {
+            annotation.setEnabled(false);
+            if (annotation.getDescription() == null || annotation.getDescription().isEmpty()) {
+                annotation.setDescription(ignore.value());
+            } else {
+                annotation.setDescription(ignore.value() + ": " + annotation.getDescription());
```

Comment:
Should we add an `empty()` check for `ignore.value()` as well, before doing this concatenation because if the user didn't specify any value for `value()`, then we will end up setting a description which looks like ` : my regression` which kind of looks odd. So I think we should add an empty check for `value` before we decide to use it to set descriptions.

Reply:
Good idea, it will do it.

Reply:
Done